hledger-lib/Hledger/Read/Common.hs

```diff
@@ -161,18 +164,23 @@ parserErrorAt pos s = throwError $ sourcePosPretty pos ++ ":\n" ++ s
 --- * parsers
 --- ** transaction bits
 
+{-# INLINABLe statusp #-}
 statusp :: TextParser m ClearedStatus
 statusp =
-  choice'
-    [ many spacenonewline >> char '*' >> return Cleared
-    , many spacenonewline >> char '!' >> return Pending
-    , return Uncleared
-    ]
-    <?> "cleared status"
+  try
+    (do many spacenonewline
+        (char '*' >> return Cleared) <|> (char '!' >> return Pending)) <|>
+  return Uncleared <?> "statusp"
```

Comment:
In general we should be trying to make parser errors more meaningful to users, this goes in the other direction